packages/frontend-app-api/api-report.md

```diff
@@ -14,16 +14,20 @@ import { SubRouteRef } from '@backstage/frontend-plugin-api';
 
 // @public (undocumented)
 export function createApp(options?: {
-  features?: FrontendFeature[];
+  features?: (FrontendFeature | CreateAppFeatureLoader)[];
   configLoader?: () => Promise<{
     config: ConfigApi;
   }>;
   bindRoutes?(context: { bind: CreateAppRouteBinder }): void;
-  featureLoader?: (ctx: { config: ConfigApi }) => Promise<FrontendFeature[]>;
 }): {
   createRoot(): JSX_2.Element;
 };
 
+// @public
+export type CreateAppFeatureLoader = (options: {
+  config: ConfigApi;
+}) => Promise<FrontendFeature[]>;
```

Comment:
Probably return `Promise<{ features: FrontendFeature[] }>` just in case right?

Regarding the Q, I think wrapping this up as a `{ loader: (...) => ... }` would make sense, this is a fairly low level API anyway.

Tbh the ability to provide a loader is here so that it's possible to experiment with dynamic plugins, and the outcome of that might be something completely different in the end anyway. But I think it's worth making it a very extensible API for now as it doesn't really need to be very lean at all.

Reply:
Yarr, fixed!